--- install.sh	2025-10-31 09:48:26.677708725 +0800
+++ install.sh	2025-11-06 10:05:09.675219575 +0800
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/bash -e
 # Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
@@ -13,8 +13,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Exit on command failure
-set -e
+export CFLAGS='-O0 -g0 '$CFLAGS CXXFLAGS='-O0 -g0 '$CXXFLAGS
+SHYNUR_CMAKE_VARS='-DSECURITY=OFF -DNO_TLS=ON '
+SHYNUR_CMAKE_VARS+=' -DSHM_TRANSPORT_DEFAULT=OFF '  # 默认不使用共享内存通信.
+SHYNUR_CMAKE_VARS+=' -DFASTDDS_STATISTICS=OFF -DSTRICT_REALTIME=OFF -DSQLITE3_SUPPORT=OFF '
+SHYNUR_CMAKE_VARS+=' -DLOG_NO_INFO=OFF -DFASTDDS_ENFORCE_LOG_INFO=ON -DLOG_NO_WARNING=OFF '
+SHYNUR_CMAKE_VARS+=' -DLOG_NO_ERROR=OFF -DINTERNAL_DEBUG=ON '
+SHYNUR_CMAKE_VARS+=' -DCOMPILE_EXAMPLES=OFF -DINSTALL_EXAMPLES=OFF -DBUILD_DOCUMENTATION=OFF '
+SHYNUR_CMAKE_VARS+=' -DCHECK_DOCUMENTATION=OFF '
+export SHYNUR_CMAKE_VARS+=" -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE:=Debug} -DCMAKE_C_FLAGS='$CFLAGS' -DCMAKE_CXX_FLAGS='$CXXFLAGS' "
+if [ 18.04 = `cat /etc/os-release | grep VERSION_ID | awk -F'"' '{print \$2}'` ]; then
+    (
+	cd /tmp
+	rm -rf asio*
+	wget https://github.com/chriskohlhoff/asio/archive/refs/tags/asio-1-12-2.tar.gz
+	tar zxf asio*.tar.gz
+	cd asio*/asio/include
+	rm -rf /usr/local/include/asio{.hpp,/}
+        mv asio.hpp asio/ /usr/local/include
+    )
+fi
 
 print_usage()
 {
@@ -122,25 +140,25 @@
     # Determine if apt-get is available
     if [[ ${INSTALL_DEPENDENCIES} == "TRUE" ]]
     then
-        # Install dependencies
-        apt update
         apt install --yes --no-install-recommends \
             git \
             build-essential \
-            cmake \
+            `#cmake` \
             libssl-dev \
-            libasio-dev \
+            `[ 18.04 != \`cat /etc/os-release | grep VERSION_ID | awk -F'"' '{print \$2}'\` ] && echo libasio-dev` \
             libtinyxml2-dev \
             openjdk-11-jre-headless \
             python3 \
-            python3-xmlschema
+            `#python3-xmlschema`
     fi
+    git config --global --add safe.directory \*
 
     # Build and install foonthan memory
     mkdir -p ${SCRIPT_DIR}/build/foonathan_memory_vendor
     cd ${SCRIPT_DIR}/build/foonathan_memory_vendor
     cmake \
-        -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} \
+        -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX}  \
+	     $SHYNUR_CMAKE_VARS  \
         ${SCRIPT_DIR}/src/foonathan_memory_vendor
     cmake --build . --target install -j ${BUILD_CORES}
     cd -
@@ -151,8 +169,9 @@
         mkdir -p ${SCRIPT_DIR}/build/fastcdr
         cd ${SCRIPT_DIR}/build/fastcdr
         cmake \
-            -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} \
-            -DBUILD_SHARED_LIBS=${SHARED_LIBS_OPTION} \
+            -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX}  \
+            -DBUILD_SHARED_LIBS=${SHARED_LIBS_OPTION}  \
+	    $SHYNUR_CMAKE_VARS  \
             ${SCRIPT_DIR}/src/fastcdr
         cmake --build . --target install -j ${BUILD_CORES}
         cd -
@@ -167,6 +186,7 @@
             -DSECURITY=${SECURITY} \
             -DCOMPILE_EXAMPLES=${INSTALL_EXAMPLES} \
             -DINSTALL_EXAMPLES=${INSTALL_EXAMPLES} \
+	    $SHYNUR_CMAKE_VARS  \
             ${SCRIPT_DIR}/src/fastdds
         cmake --build . --target install -j ${BUILD_CORES}
         cd -
