--- install.sh	2025-10-31 09:48:26.677708725 +0800
+++ install.sh	2025-10-31 10:21:42.443081398 +0800
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/bash -e
 # Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
@@ -13,8 +13,13 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Exit on command failure
-set -e
+: ${CMAKE_BUILD_TYPE:=Debug}
+export CFLAGS='-O0 -g3 '$CFLAGS CXXFLAGS='-O0 -g3 '$CXXFLAGS
+SHYNUR_CMAKE_VARS='-DSECURITY=OFF -DNO_TLS=ON '
+SHYNUR_CMAKE_VARS+=' -DSHM_TRANSPORT_DEFAULT=ON '  # 默认使用共享内存通信.
+SHYNUR_CMAKE_VARS+=' -DFASTDDS_STATISTICS=OFF -DSTRICT_REALTIME=OFF -DSQLITE3_SUPPORT=OFF '
+SHYNUR_CMAKE_VARS+=' -DLOG_NO_INFO=OFF -DFASTDDS_ENFORCE_LOG_INFO=OFF -DLOG_NO_WARNING=OFF '
+SHYNUR_CMAKE_VARS+=' -DLOG_NO_ERROR=OFF -DINTERNAL_DEBUG=ON '
 
 print_usage()
 {
@@ -122,25 +127,25 @@
     # Determine if apt-get is available
     if [[ ${INSTALL_DEPENDENCIES} == "TRUE" ]]
     then
-        # Install dependencies
-        apt update
         apt install --yes --no-install-recommends \
             git \
             build-essential \
-            cmake \
+            #cmake \
             libssl-dev \
             libasio-dev \
             libtinyxml2-dev \
             openjdk-11-jre-headless \
             python3 \
-            python3-xmlschema
+            #python3-xmlschema
     fi
+    git config --global --add safe.directory \*
 
     # Build and install foonthan memory
     mkdir -p ${SCRIPT_DIR}/build/foonathan_memory_vendor
     cd ${SCRIPT_DIR}/build/foonathan_memory_vendor
     cmake \
         -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} \
+	$SHYNUR_CMAKE_VARS  \
         ${SCRIPT_DIR}/src/foonathan_memory_vendor
     cmake --build . --target install -j ${BUILD_CORES}
     cd -
@@ -152,6 +157,7 @@
         cd ${SCRIPT_DIR}/build/fastcdr
         cmake \
             -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} \
+	    $SHYNUR_CMAKE_VARS  \
             -DBUILD_SHARED_LIBS=${SHARED_LIBS_OPTION} \
             ${SCRIPT_DIR}/src/fastcdr
         cmake --build . --target install -j ${BUILD_CORES}
@@ -164,6 +170,7 @@
             -DCMAKE_PREFIX_PATH=${INSTALL_PREFIX} \
             -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} \
             -DBUILD_SHARED_LIBS=${SHARED_LIBS_OPTION} \
+	    $SHYNUR_CMAKE_VARS  \
             -DSECURITY=${SECURITY} \
             -DCOMPILE_EXAMPLES=${INSTALL_EXAMPLES} \
             -DINSTALL_EXAMPLES=${INSTALL_EXAMPLES} \
